api/v1alpha4/types.go

```diff
@@ -160,11 +164,12 @@ type IngressRules []IngressRule
 
 // LoadBalancerSpec defines an Azure load balancer.
 type LoadBalancerSpec struct {
-	ID          string       `json:"id,omitempty"`
-	Name        string       `json:"name,omitempty"`
-	SKU         SKU          `json:"sku,omitempty"`
-	FrontendIPs []FrontendIP `json:"frontendIPs,omitempty"`
-	Type        LBType       `json:"type,omitempty"`
+	ID               string       `json:"id,omitempty"`
+	Name             string       `json:"name,omitempty"`
+	SKU              SKU          `json:"sku,omitempty"`
+	FrontendIPs      []FrontendIP `json:"frontendIPs,omitempty"`
+	Type             LBType       `json:"type,omitempty"`
+	FrontendIPsCount *int32       `json:"frontendIPsCount,omitempty"`
```

Comment:
I could see this getting confusing for users, the list of frontendIPs above is redundant information with `FrontendIPsCount`... need to think more about this 🤔

Reply:
Yeah, I was earlier thinking something like `minFrontendIPsRequired`, and add some comments to make it clear that this field will not be used when frontendIps are configured. Also, instead of modifying `LoadBalancerSpec` to include FrontendIPsCount, maybe we could create a new type specific to NodeOutboundLb that composes LoadBalacerSpec with this new field. wdyt?

Reply:
Another UX I don't like is the user having to specify an empty spec for node outbound lb for private cluster. Not sure how to avoid it though since we want to allow disabling node outbound LB.

Reply:
> Another UX I don't like is the user having to specify an empty spec for node outbound lb for private cluster. Not sure how to avoid it though since we want to allow disabling node outbound LB.

I agree. We would need a specific "disable" field to make it explicit. I'm leaning towards separating the two types. @devigned @nader-ziada thoughts?

Reply:
> add some comments to make it clear that this field will not be used when frontendIps are configured

this should also be enforced by validation. I think we can allow both as long as they are consistent with one another, or make them mutually exclusive completely.

Reply:
> this should also be enforced by validation. I think we can allow both as long as they are consistent with one another, or make them mutually exclusive completely.

Since we're just going to enable `frontendIPsCount` to be mutated, this shouldn't be an issue at least for now.  I agree though that this should be made mutually exclusive once we open up other options.

Reply:
sorry for the late reply, missed this comment earlier, agreed its confusing, 

I agree that a new type for `NodeOutboundLbSpec` that would include the count (as mnetioned above) is much easier to understand, even if there duplication of fields with `LoadBalancerSpec`

and should be enforced by  validation